proxytools/scrappers/premproxy.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import logging
+import re
+import jsbeautifier.unpackers.packer as packer
+
+from bs4 import BeautifulSoup
+
+from ..proxy_scrapper import ProxyScrapper
+from ..utils import validate_ip
+
+log = logging.getLogger(__name__)
+
+
+# This site has a "bot" network checker and may result in the
+# pages not being loaded. This should only happen if you request
+# scans too frequently.
+class Premproxy(ProxyScrapper):
+
+    def __init__(self, args):
+        super(Premproxy, self).__init__(args, 'premproxy-com')
+        self.base_url = 'https://premproxy.com'
+
+    def scrap(self):
+        proxylist = []
+        urls = self.extract_pages()
+        print urls
+        for url in urls:
+            html = self.request_url(url)
+            if html is None:
+                log.error('Failed to download webpage: %s', url)
+                continue
+
+            log.info('Parsing webpage from: %s', url)
+            proxylist.extend(self.parse_webpage(html))
+
+        return proxylist
+
+    def parse_webpage(self, html):
+        proxylist = []
+        soup = BeautifulSoup(html, 'html.parser')
+        # soup.prettify()
+
+        # For now, we'll assume <script> #2 in the list is the one with ports.
+        scripts = soup.findAll('script')

```

Comment:
Instead of relying on having the second script with the `eval(function(p,a,c,k,e,d` maybe we could request all scripts and test if it starts with: `eval(function(p,a,c,k,e,d`. If yes, then proceed to unpack it. This is only for robustness and shouldn't be a priority.